cool-demos/conditional-blocks.html

```diff
@@ -0,0 +1,439 @@
+                                { value: 'name_of_event', text: 'name_of_event' },
+                                { value: 'length_of_stay', text: 'length_of_stay' },
+                            ]
+                        }, {
+                            type: 'selectbox',
+                            name: 'operator',
+                            label: 'Operator',
+                            items: [
+                                { value: 'is greater than', text: 'is greater than' },
+                                { value: 'is equal or greater than', text: 'is equal or greater than' },
+                                { value: 'is equal to', text: 'is equal to' },
+                                { value: 'is equal or less than', text: 'is equal or less than' },
+                                { value: 'is less than', text: 'is less than' },
+                                { value: 'is not equal to', text: 'is not equal to' },
+                            ]
+                        }, {
+                            type: 'input',
+                            name: 'value',
+                            label: 'Value',
+                            placeholder: 'Value'
+                        }
+                    ]
+                },
+                buttons: [
+                {
+                    type: 'cancel',
+                    name: 'closeButton',
+                    text: 'Cancel'
+                },
+                {
+                    type: 'submit',
+                    name: 'submitButton',
+                    text: 'Save',
+                    primary: true
+                }
+                ],
+                initialData: {
+                    property: conditionalBlock ? conditionalBlock.dataset.property : 'number_of_people' ,
+                    operator: conditionalBlock ? conditionalBlock.dataset.operator : 'is equal to',
+                    value: conditionalBlock ? conditionalBlock.dataset.value: ''
+                },
+                onSubmit: (dialog) => {
+                    // Get the form data.
+                    var data = dialog.getData();
+
+                    // Check if a block is edited or a new block is to be inserted
+                    if (!conditionalBlock) {
+                        // Insert content at the location of the cursor.
+                        // https://www.tiny.cloud/docs/api/tinymce/tinymce.editor/#insertcontent
+                        editor.insertContent(`<conditional-block data-property="${data.property}" data-operator="${data.operator}" data-value="${data.value}"><p>Write conditional text here</p></conditional-block>`);
+                    }
+                    else {
+                        // Get the currently selected node.
+                        // https://www.tiny.cloud/docs/api/tinymce.dom/tinymce.dom.selection/
+                        const node = editor.selection.getNode();
```

Comment:
Should we be operating on the currently selected node or should we be finding a way to get the node from `conditionalBlock` to operate on instead? This only matters if there is a way to open this dialog by passing in a `conditionalBlock` that isn't the currently selected node.

Reply:
It seems like I'm not using the node constant anywhere, and it all works if I remove that line.